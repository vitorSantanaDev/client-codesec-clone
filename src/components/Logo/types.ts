export type IViewProps = {
  srcImage: string
  altText: string
  url: string
  className?: string
}
