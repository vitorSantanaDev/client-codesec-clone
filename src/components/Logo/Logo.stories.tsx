import { ComponentStory, ComponentMeta } from '@storybook/react'
import Logo from '.'

export default {
  title: 'Logo',
  component: Logo,
  args: {
    url: '/',
    srcImage: '/images/logo-2-codesec.svg',
    altText: 'Logo codesec'
  }
} as ComponentMeta<typeof Logo>

export const Default: ComponentStory<typeof Logo> = (args) => <Logo {...args} />

export const SecondaryLogo: ComponentStory<typeof Logo> = (args) => (
  <Logo {...args} />
)

SecondaryLogo.args = {
  srcImage: '/images/codesec-logo.svg'
}
