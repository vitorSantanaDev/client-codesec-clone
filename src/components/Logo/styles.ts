import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css``}
`
export const Logo = styled.img`
  ${() => css`
    cursor: pointer;
  `}
`
