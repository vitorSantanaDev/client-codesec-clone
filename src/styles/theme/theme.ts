export const theme = {
  colors: {
    redPrimary: '#E83752',
    redLight: '#FA5770',
    redDark: '#D4203B',
    redError: '#EB3D3D',
    redGradient: 'linear-gradient(243.46deg, #FA5770 16.66%, #E83752 83.34%)',
    redGradientDark:
      'linear-gradient(225.05deg, #FA5770 0%, rgba(232, 55, 82, 0) 100%)',
    redGradientLight:
      'linear-gradient(225.05deg, rgba(250, 87, 112, 0.1) 0%, rgba(232, 55, 82, 0) 50%)',
    greenSucsses: '#32D957',
    white: '#FFFFFF',
    whiteLight: '#F5F0F1',
    whiteDark: '#EDE4E5',
    black: '#121111',
    blackLight: '#1C1B1B',
    blackDark: '#0A0909',
    gray1: '#141414',
    gray2: '#2C2C2C',
    gray3: '#444444',
    gray4: '#5C5C5C',
    gray5: '#747474',
    gray6: '#8C8C8C',
    gray7: '#A3A3A3'
  },
  font: {
    family: {
      default: "'Raleway', sans-serif"
    },
    sizes: {
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
      xxhuge: '8rem'
    }
  },
  media: {
    lteMedium: '(max-width: 768px)',
    lteLarge: '(max-width: 820px)',
    ltexLarge: '(max-width: 912px)',
    lteSmall: '(max-width: 420px)',
    lteXSmall: '(max-width: 280px)'
  },
  spacings: {
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
    xxhuge: '8rem'
  }
}
