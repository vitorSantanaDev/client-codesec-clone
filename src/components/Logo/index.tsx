import Link from 'next/link'
import * as S from './styles'

type LogoProps = {
  srcImage: string
  altText: string
  url: string
}

const Logo = ({ srcImage, altText, url }: LogoProps) => (
  <S.Wrapper>
    <Link href={url}>
      <S.Logo src={srcImage} alt={altText} />
    </Link>
  </S.Wrapper>
)

export default Logo
