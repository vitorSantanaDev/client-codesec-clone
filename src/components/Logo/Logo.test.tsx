import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import Logo from '.'

describe('<Logo />', () => {
  it('should render logo codesec', () => {
    const { container } = renderTheme(
      <Logo
        url="/"
        srcImage="/images/codesec-logo.svg"
        altText="Logo codesec"
      />
    )
    const logoImage = screen.getByAltText(/logo codesec/i)
    expect(logoImage).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
