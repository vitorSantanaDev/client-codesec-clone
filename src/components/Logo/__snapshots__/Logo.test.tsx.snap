// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render logo codesec 1`] = `
.c0 {
  cursor: pointer;
}

<div
  class=""
>
  <img
    alt="Logo codesec"
    class="c0"
    src="/images/codesec-logo.svg"
  />
</div>
`;
